### PR TITLE
All solutions now compile properly, including tests [dev3x]

### PR DESCRIPTION
**Pull request by MarkZuber**

### CODE REVIEW

File: tests/devapps/DesktopTestApp/MsalUserRefreshTokenControl.cs [hunk 1]
```diff
@@ -24,14 +24,14 @@ public partial class MsalUserRefreshTokenControl : UserControl
 
         internal MsalUserRefreshTokenControl(PublicClientApplication publicClient, MsalRefreshTokenCacheItem rtIitem) : this()
         {
-            this.publicClient = publicClient;
-            cache = publicClient.UserTokenCache;
-            rtItem = rtIitem;
+            _publicClient = publicClient;
+            _cache = publicClient.UserTokenCacheInternal;
+            _rtItem = rtIitem;
```
Comment (jennyf19) [inline tests/devapps/DesktopTestApp/MsalUserRefreshTokenControl.cs]: >rtIitem [](start = 22, length = 7)

sp?

File: tests/devapps/DesktopTestApp/MsalUserRefreshTokenControl.cs [hunk 2]
```diff
@@ -24,14 +24,14 @@ public partial class MsalUserRefreshTokenControl : UserControl
...
 
-            accountItem = cache.GetAccount(rtIitem, new RequestContext(null, new MsalLogger(Guid.NewGuid(), null)));
+            accountItem = _cache.GetAccount(rtIitem, RequestContext.CreateForTest());
```
Comment (jennyf19) [inline tests/devapps/DesktopTestApp/MsalUserRefreshTokenControl.cs]: >r [](start = 44, length = 1)

should this be _rtItem?

File: tests/devapps/DesktopTestApp/PublicClientHandler.cs [hunk 3]
```diff
@@ -38,11 +40,11 @@ class PublicClientHandler
         public PublicClientHandler(string clientId)
         {
             ApplicationId = clientId;
-            PublicClientApplication = new PublicClientApplication(ApplicationId)
-            {
-                UserTokenCache = TokenCacheHelper.GetUserCache(),
-                Component = _component
-            };
+            PublicClientApplication = PublicClientApplicationBuilder.Create(ApplicationId)
+                .WithComponent(_component)
```
Comment (jennyf19) [inline tests/devapps/DesktopTestApp/PublicClientHandler.cs]: >WithComponent [](start = 17, length = 13)

What is WithComponent?

File: tests/Microsoft.Identity.Test.Unit.net45/JsonWebTokenTests.cs [hunk 4]
```diff
@@ -88,31 +89,21 @@ internal void SetupMocks(MockHttpManager httpManager)
...
-                    _cache,
-                    _cache)
-                {
-                    ValidateAuthority = false
-                };
+                var app = ConfidentialClientApplicationBuilder.Create(MsalTestConstants.ClientId)
+                                                              .AddKnownAuthority(
+                                                                  new System.Uri(ClientApplicationBase.DefaultAuthority),
+                                                                  true).WithRedirectUri(MsalTestConstants.RedirectUri)
+                                                              .WithX509Certificate2(certificate).BuildConcrete();
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Unit.net45/JsonWebTokenTests.cs]: >WithX509Certificate2 [](start = 63, length = 20)

this looks ugly to me. can we change the name to something more generic?

File: tests/Microsoft.Identity.Test.Unit.net45/CacheTests/TokenCacheTests.cs [hunk 5]
```diff
@@ -50,14 +50,14 @@ public class TokenCacheTests
         public static long ValidExpiresIn = 3600;
         public static long ValidExtendedExpiresIn = 7200;
 
-        private TokenCache _cache;
+        //private TokenCache _cache;
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Unit.net45/CacheTests/TokenCacheTests.cs]: >// [](start = 8, length = 2)

this code is soooo dead.

File: tests/Microsoft.Identity.Test.Unit.net45/CoreTests/CacheTests/CacheFallbackOperationsTests.cs [hunk 6]
```diff
@@ -43,13 +43,15 @@ namespace Microsoft.Identity.Test.Unit.CoreTests.CacheTests
...
         private InMemoryLegacyCachePersistence _legacyCachePersistence;
+        private ICoreLogger _logger;
 
         [TestInitialize]
         public void TestInitialize()
         {
             // Methods in CacheFallbackOperations silently catch all exceptions and log them;
             // By setting this to null, logging will fail, making the test fail.
-            MsalLogger.Default = Substitute.For<ICoreLogger>();
+            //MsalLogger.Default = Substitute.For<ICoreLogger>();
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Unit.net45/CoreTests/CacheTests/CacheFallbackOperationsTests.cs]: >// [](start = 12, length = 2)

dead code?

File: tests/devapps/DesktopTestApp/PublicClientHandler.cs (hunk 3 above)
Comment (MarkZuber) [reply tests/devapps/DesktopTestApp/PublicClientHandler.cs]: This is the component name that apps can set to differentiate themselves in telemetry.  It's a terrible name, but it's currently a mirror for the Component property on clientapplicationbase.  We should review this as part of the updated public api surface.

---
In reply to: [247743496](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/786#discussion_r247743496) [](ancestors = 247743496)

File: tests/Microsoft.Identity.Test.Unit.net45/JsonWebTokenTests.cs (hunk 4 above)
Comment (MarkZuber) [reply tests/Microsoft.Identity.Test.Unit.net45/JsonWebTokenTests.cs]: I'll talk with jm on this with the public api surface review.  i've added this to the discussion list.

---
In reply to: [247743987](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/786#discussion_r247743987) [](ancestors = 247743987)

**Pull request merged**